### PR TITLE
lua_api.txt: Various fixes and improvements

### PR DESCRIPTION
**Pull request by pauloue**
See diff for the first commit:

* improve item types descriptions (can probably check the box in #5854 but the associated issue may be more than documentation)
* add replacements field to fuel def using lava bucket as example
* remove an extra example (there's another one above that's very similar)

Add blank lines after headers for readability and consistency

Linkify section references: replace quotes around section names with brackets so they can be linked when compiled to HTML. Works with pandoc which I use here: [pauloue.gitlab.io/minetest-modding-api](https://pauloue.gitlab.io/minetest-modding-api/)

### CODE REVIEW

File: doc/lua_api.txt [hunk 1]
```diff
@@ -5636,7 +5860,7 @@ Used by `minetest.register_node`.
         -- Only when `paramtype2` supports palettes.
 
         post_effect_color = "green#0F",
-        -- Screen tint if player is inside node, see "ColorSpec".
+        -- Screen tint if player is inside node, see "ColorSpec"
```
[the remark this answers is not in the document]
Comment (pauloue) [reply doc/lua_api.txt]: It doesn't work in a code block

File: doc/lua_api.txt [hunk 2]
```diff
@@ -4580,13 +4767,14 @@ automatically chosen for you.
...
-      {
-        enabled = boolean,     -- whether to enable, default true
-        block_radius = number, -- the radius (in nodes) of the areas the cache
-                                  generates prefiltered lists for, minimum 16,
-                                  default 64.
-        limit = number,        -- the cache's size, minimum 20, default 1000
-      }
+
+          {
+            enabled = boolean,      -- Whether to enable, default true
```
[the remark this answers is not in the document]
Comment (pauloue) [reply doc/lua_api.txt]: I removed the braces completely to fit it in 80 cols

**Comment by ClobberXD** [comment]
Links seem to require `(<actual link>)` after `[title]`, at least on Github. Links with only the links enclosed in `[` and `]` aren't parsed as hyperlinks at all. See [this gist](https://gist.github.com/ClobberXD/11f4d8e1995ea187e095c03b79dfdb5a) (and it's revisions) using which I've experimented with various link formats.

**Comment by pauloue** [comment]
GitHub markdown doesn't support this syntax, [pandoc](http://pandoc.org/MANUAL.html#extension-implicit_header_references) is the only one I know of that does. I think it would be best to avoid using a more verbose link syntax that wouldn't make sense in txt form.

With pandoc you can do:

    pandoc --toc -s --indented-code-classes lua -o lua-api.html lua_api.txt

and you get a webpage with a table of contents, lua syntax highlighting, and the internal links.

**Comment by ClobberXD** [comment]
Sorry if I've misunderstood, but if this is pandoc-specific syntax which isn't supported by most other markdown parsers, I feel that this PR shouldn't be merged without removing these...

**Comment by pauloue** [comment]
A script could be used to add header IDs after the brackets for other parsers if necessary, but why not just use pandoc if internal links are wanted? Having the header IDs in txt form would be a bit messy.

**Comment by paramat** [comment]
Spaces after section titles are fine for me, makes the 3rd level titles clearer.

File: doc/lua_api.txt [hunk 3]
```diff
@@ -754,34 +807,34 @@ Examples of sound parameter tables:
 
     -- Play locationless on all clients
     {
-        gain = 1.0, -- default
-        fade = 0.0, -- default, change to a value > 0 to fade the sound in
-        pitch = 1.0, -- default
+        gain = 1.0,  -- default
+        fade = 0.0,  -- default, change to a value > 0 to fade the sound in
+        pitch = 1.0,  -- default
```
[the remark this answers is not in the document]
Comment (paramat) [reply doc/lua_api.txt]: I think 2 spaces before the comment is intentional.

File: doc/lua_api.txt [hunk 4]
```diff
@@ -1312,10 +1298,11 @@ Item types
...
 
-* Node (`register_node`): A node from the world.
-* Tool (`register_tool`): A tool/weapon that can dig and damage
-  things according to `tool_capabilities`.
-* Craftitem (`register_craftitem`): A miscellaneous item.
+* Node: An item that can be placed in the world as a node.
+* Tool: An item that can be worn out and repaired, cannot be stacked, and often
+  digs or deals damage.
+* Craftitem: A general purpose item that can be used in crafting recipes, eaten,
+  placed, or used on things in the world.
```
Comment (paramat) [inline doc/lua_api.txt]: Not sure about 'placed', maybe 'dropped'?

File: doc/lua_api.txt [hunk 5]
```diff
@@ -1692,8 +1679,8 @@ Metadata
 Node Metadata
 -------------
 The instance of a node in the world normally only contains the three values
-mentioned in "Nodes". However, it is possible to insert extra data into a
-node. It is called "node metadata"; See `NodeMetaRef`.
+mentioned in "Node paramtypes". However, it is possible to insert extra data
```
Comment (paramat) [inline doc/lua_api.txt]: "Nodes" is correct, this is referring to
`{name="name", param1=num, param2=num}`

File: doc/lua_api.txt [hunk 6]
```diff
@@ -4579,14 +4564,13 @@ automatically chosen for you.
...
-    * `params`:
-      {
-        enabled = boolean,     -- whether to enable, default true
-        block_radius = number, -- the radius (in nodes) of the areas the cache
-                                  generates prefiltered lists for, minimum 16,
-                                  default 64.
-        limit = number,        -- the cache's size, minimum 20, default 1000
-      }
+    * `params` is a table with the following fields:
+
```
Comment (paramat) [inline doc/lua_api.txt]: Empty line seems wrong.

**Comment by paramat** [comment]
First commit is fine apart from line comments.

File: doc/lua_api.txt [hunk 7]
```diff
@@ -201,7 +201,7 @@ wants to register. Subsequent execution depends on minetest calling the
 registered callbacks.
 
 `minetest.settings` can be used to read custom or existing settings at load
-time, if necessary. (See `Settings`)
+time, if necessary. (See [`Settings`])
```
Comment (paramat) [inline doc/lua_api.txt]: No ``

File: doc/lua_api.txt [hunk 8]
```diff
@@ -1764,8 +1764,8 @@ Node Metadata
 -------------
 
 The instance of a node in the world normally only contains the three values
-mentioned in "Node paramtypes". However, it is possible to insert extra data
-into a node. It is called "node metadata"; See `NodeMetaRef`.
+mentioned in [Node paramtypes]. However, it is possible to insert extra data
```
Comment (paramat) [inline doc/lua_api.txt]: [Nodes]

File: doc/lua_api.txt [hunk 9]
```diff
@@ -1808,7 +1808,7 @@ Example:
 Item Metadata
 -------------
 
-Item stacks can store metadata too. See `ItemStackMetaRef`.
+Item stacks can store metadata too. See [`ItemStackMetaRef`].
```
Comment (paramat) [inline doc/lua_api.txt]: No ``

File: doc/lua_api.txt [hunk 10]
```diff
@@ -3035,7 +3035,7 @@ and the array index for a position p contained completely in p1..p2 is:
 
 Note that this is the same "flat 3D array" format as
 `PerlinNoiseMap:get3dMap_flat()`.
-VoxelArea objects (see section 'VoxelArea') can be used to simplify calculation
+VoxelArea objects (see section [`VoxelArea`]) can be used to simplify calculation
```
Comment (paramat) [inline doc/lua_api.txt]: No ``

File: doc/lua_api.txt (hunk 9 above)
Comment (pauloue) [reply doc/lua_api.txt]: This is referring to the section in the class reference which does have the backticks, so it must be the same to be linked.

**Comment by paramat** [comment]
2nd commit is fine.
3rd commit fine apart from line comment.
+1 once issues resolved.

File: doc/lua_api.txt (hunk 4 above)
Comment (pauloue) [reply doc/lua_api.txt]: Items can be placed; doors in MTG are craftitems. And everything can be dropped.

File: doc/lua_api.txt (hunk 6 above)
Comment (paramat) [reply doc/lua_api.txt]: Ah ok, fine then.

File: doc/lua_api.txt (hunk 4 above)
Comment (paramat) [reply doc/lua_api.txt]: Special code places door nodes when a door craftitem is 'placed'. Craftitems can't naturally be placed like nodes, only dropped as objects.

File: doc/lua_api.txt (hunk 6 above)
Comment (paramat) [reply doc/lua_api.txt]: I mean, it's fine as it is.

File: doc/lua_api.txt (hunk 5 above)
Comment (paramat) [reply doc/lua_api.txt]: Now fixed.

File: doc/lua_api.txt (hunk 8 above)
Comment (paramat) [reply doc/lua_api.txt]: Now fixed.

File: doc/lua_api.txt (hunk 4 above)
Comment (pauloue) [reply doc/lua_api.txt]: ok

Comment (paramat) [reply doc/lua_api.txt]: Now fixed.

File: doc/lua_api.txt [hunk 11]
```diff
@@ -754,34 +807,34 @@ Examples of sound parameter tables:
...
-        pitch = 1.0, -- default
+        gain = 1.0,   -- default
+        fade = 0.0,   -- default, change to a value > 0 to fade the sound in
+        pitch = 1.0,  -- default
     }
     -- Play locationless to one player, looped
     {
         to_player = name,
-        gain = 1.0, -- default
+        gain = 1.0,  -- default
```
Comment (SmallJoker) [inline doc/lua_api.txt]: Totally fine to line-up, but why do you use one space more than necessary in these examples?

Comment (pauloue) [reply doc/lua_api.txt]: I was going for 2 spaces separation from the code as suggested by the Lua style guide

Comment (SmallJoker) [reply doc/lua_api.txt]: Alright, I'm wrong in this point -- didn't know it was set to two spaces.

**Comment by SmallJoker** [comment]
@pauloue Thanks. Paramat or me will update it if there are any further concerns - hopefully it will not need rebasing.

File: doc/lua_api.txt [hunk 12]
```diff
@@ -1310,15 +1363,18 @@ Items
...
 
-* Node (`register_node`): A node from the world.
-* Tool (`register_tool`): A tool/weapon that can dig and damage
-  things according to `tool_capabilities`.
-* Craftitem (`register_craftitem`): A miscellaneous item.
+* Node: An item that can be placed in the world as a node.
+* Tool: An item that can be worn out and repaired, cannot be stacked, and often
+  digs or deals damage.
+* Craftitem: A general purpose item that can be used in crafting recipes, eaten,
+  dropped, or used on things in the world.
```
[the remark this answers is not in the document]
Comment (paramat) [reply doc/lua_api.txt]: I'm also unsure about 'general purpose'.

File: doc/lua_api.txt [hunk 13]
```diff
@@ -93,29 +97,28 @@ Mods
...
+World-specific games
+--------------------
+
 It is possible to include a game in a world; in this case, no mods or
 games are loaded or checked from anywhere else.
 
 This is useful for e.g. adventure worlds.
 
-This happens if the following directory exists:
+This happens if the `<worldname>/game/` directory exists.
```
[the remark this answers is not in the document]
Comment (paramat) [reply doc/lua_api.txt]: 'as soon as' -> 'if'.

[the remark this answers is not in the document]
Comment (paramat) [reply doc/lua_api.txt]: Now done and fine for me.

**Comment by paramat** [comment]
Only the craftitem text needs work but we can do that on merge.

File: doc/lua_api.txt (hunk 12 above)
[the remark this answers is not in the document]
Comment (paramat) [reply doc/lua_api.txt]: We decided to clean this up later.

**Comment by paramat** [comment]
Note: Craftitem description needs improving in a follow-up PR.
@pauloue

File: doc/lua_api.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (Bituvo) [reply doc/lua_api.txt:5860]: Yes it does.

**Pull request merged**